Makefile

```diff
@@ -110,35 +124,44 @@ e2e-local:
 		$(DEBUG) --go-test-flags "$(GO_TEST_FLAGS)"
 
 .PHONY: run
-run:
-	operator-sdk up local \
+run: ## Run the operator outside of a cluster in development mode
+	operator-sdk run --local \
 		--go-ldflags=$(LDFLAGS) \
-		--namespace=$(RUN_NAMESPACE) \
+		--watch-namespace=$(RUN_NAMESPACE) \
 		--operator-flags="-dev"
 
 .PHONY: demo
-demo:
-	operator-sdk up local \
+demo: ## Run the operator outside of a cluster using the demo driver
+	operator-sdk run --local \
 		--go-ldflags=$(LDFLAGS) \
-		--namespace=$(RUN_NAMESPACE) \
+		--watch-namespace=$(RUN_NAMESPACE) \
 		--operator-flags="-dev -demo-mode"
 
 .PHONY: docker
-docker:
+docker: docker-operator docker-sdk docker-golint ## Build docker images
+
+.PHONY: docker-operator
+docker-operator:
 	docker build . -f build/Dockerfile
 
+.PHONY: docker-sdk
+docker-sdk:
+	docker build . -f hack/Dockerfile.operator-sdk
+
+.PHONY: docker-golint
+docker-golint:
+	docker build . -f hack/Dockerfile.golint
+
 .PHONY: build
-build:
+build: ## Build the operator binary
 	@echo LDFLAGS=$(LDFLAGS)
-	go build -mod=vendor -o build/_output/bin/baremetal-operator cmd/manager/main.go
+	go build -mod=vendor -ldflags $(LDFLAGS) -o build/_output/bin/baremetal-operator cmd/manager/main.go
 
 .PHONY: tools
 tools:
-	@echo LDFLAGS=$(LDFLAGS)
-	go build -mod=vendor -o build/_output/bin/get-hardware-details cmd/get-hardware-details/main.go
+	go build -o build/_output/bin/get-hardware-details cmd/get-hardware-details/main.go
```

Comment:
@honza This broke the OSE build -- how do we make sure your script preserves this?

Reply:
Good catch, I'll fix it ASAP.

This will go away if we upgrade to Golang 1.14. From the docs:

> When the main module contains a top-level vendor directory and its go.mod file specifies go 1.14 or higher, the go command now defaults to -mod=vendor for operations that accept that flag. A new value for that flag, -mod=mod, causes the go command to instead load modules from the module cache (as when no vendor directory is present). 

I'll see about upgrading the module to 1.14.

Reply:
#76

Reply:
https://github.com/metal3-io/baremetal-operator/pull/557